### PR TITLE
TST: Series.update with categorical

### LINKED ISSUE #25744: Series.update fails with categorical types
```python
cats = pandas.api.types.CategoricalDtype(['a', 'b', 'c', 'd'])
# cats = None
s1 = pandas.Series(['a', 'b', 'c'], index=[1, 2, 3], dtype=cats)
s2 = pandas.Series(['b', 'a'], index=[1, 2], dtype=cats)

s1.update(s2)
```

With the dtype=None, the `s1` series is updated to be `['b', 'a', 'c']` but with the `dtype=cats`, the update fails with error "ValueError: NumPy boolean array indexing assignment cannot assign 3 input values to the 2 output values where the mask is true"

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Linux
OS-release: 4.15.0-36-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 3.3.2
pip: 18.0
setuptools: 40.5.0
Cython: 0.27.3
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.6.6
patsy: 0.5.1
dateutil: 2.7.2
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.4
feather: None
matplotlib: 3.0.1
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml.etree: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### LINKED ISSUE #27338: CategoricalDtype.update_dtype fails with Categorical
ref https://github.com/pandas-dev/pandas/pull/27318#discussion_r302265250 it appears that CategoricalDtype might want to accept a Categorical as an argument but fails to handle that properly. Not sure intent here but we either want to fix to allow Categorical or alternately simplify and only allow `str/CategoricalDtype` input

### PR DESCRIPTION
**Pull request by ftrihardjo**
- [ ] closes #25744, closes #27338
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Hi @ftrihardjo - could you please put a meaningful title and link to the issue you're closing in the pull request body?

File: pandas/tests/series/methods/test_update.py [hunk 1]
```diff
@@ -108,3 +108,11 @@ def test_update_from_non_series(self, series, other, expected):
     def test_update_extension_array_series(self, result, target, expected):
         result.update(target)
         tm.assert_series_equal(result, expected)
+    def test_update_with_categorical_type(self):
```
Comment (jreback) [inline pandas/tests/series/methods/test_update.py:111]: can you add a blank line before the test.

File: pandas/tests/series/methods/test_update.py [hunk 2]
```diff
@@ -108,3 +108,11 @@ def test_update_from_non_series(self, series, other, expected):
...
+        # GH 25744
+        t = CategoricalDtype(['a', 'b', 'c', 'd'])
```
Comment (jreback) [inline pandas/tests/series/methods/test_update.py:113]: call this dtype

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, ping on green,

**Comment by jreback** [comment]
see here: https://github.com/pandas-dev/pandas/pull/38873/checks?check_run_id=1634842619

make sure you have `precommit` installed it will fix anything when you commit

**Review by MarcoGorelli — APPROVED** [review]
Thanks @ftrihardjo !

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ftrihardjo

**Pull request merged**